roofit/roofitcore/src/RooMinimizer.cxx

```diff
@@ -96,43 +97,81 @@ void RooMinimizer::cleanup()
+RooMinimizer::RooMinimizer(RooAbsReal &function, FcnMode fcnMode) : _cfg()
+{
+   // Config setting
+   _cfg.fcnMode = fcnMode;
 
-   switch (_fcnMode) {
+   // Call init functions
+   initMinimizerFirstPart();
+   initSerial(function);
+   initMinimizerFcnDependentPart(function.defaultErrorLevel());
+};
+
+/// Initialize the minimizer with old style likelihoods
+void RooMinimizer::initSerial(RooAbsReal &function)
+{
+   switch (_cfg.fcnMode) {
    case FcnMode::classic: {
-      _fcn = new RooMinimizerFcn(&function, this, _verbose);
+      _fcn = std::make_unique<RooMinimizerFcn>(RooMinimizerFcn(&function, this, _cfg.verbose));
       break;
    }
    case FcnMode::gradient: {
-      _fcn = new RooGradMinimizerFcn(&function, this, _verbose);
+      _fcn = std::make_unique<RooGradMinimizerFcn>(RooGradMinimizerFcn(&function, this, _cfg.verbose));
       break;
    }
-   case FcnMode::generic_wrapper : {
+   case FcnMode::generic_wrapper: {
       throw std::logic_error("In RooMinimizer constructor: fcnMode::generic_wrapper cannot be used on a RooAbsReal! "
                              "Please use the TestStatistics::RooAbsL based constructor instead.");
    }
    default: {
       throw std::logic_error("In RooMinimizer constructor: fcnMode has an unsupported value!");
    }
    }
-
-   initMinimizerFcnDependentPart(function.defaultErrorLevel());
 }
 
-RooMinimizer::RooMinimizer(std::shared_ptr<RooFit::TestStatistics::RooAbsL> likelihood,
-                           RooFit::TestStatistics::LikelihoodMode likelihoodMode,
-                           RooFit::TestStatistics::LikelihoodGradientMode likelihoodGradientMode)
-   : _fcnMode(FcnMode::generic_wrapper)
+/// Initialize the minimizer with new style likelihoods
+void RooMinimizer::initMultiProcess(std::shared_ptr<RooFit::TestStatistics::RooAbsL> &function)
 {
-   initMinimizerFirstPart();
-   _fcn = new RooFit::TestStatistics::MinuitFcnGrad(likelihood, this, _theFitter->Config().ParamsSettings(), likelihoodMode,
-                                                    likelihoodGradientMode, _verbose);
-   initMinimizerFcnDependentPart(likelihood->defaultErrorLevel());
-}
+#ifdef R__HAS_ROOFIT_MULTIPROCESS
+   RooFit::MultiProcess::Config::setDefaultNWorkers(_cfg.nWorkers);
+   
+   if (!_cfg.parallelGradient)
+      coutI(InputArguments) << "New style likelihood detected, setting parallel gradient calculation mode."
+                            << std::endl;
 
-// constructor helpers
+   _cfg.fcnMode = FcnMode::generic_wrapper;
+   _fcn = std::make_unique<RooFit::TestStatistics::MinuitFcnGrad>(
+      RooFit::TestStatistics::MinuitFcnGrad(
+      function, this, _theFitter->Config().ParamsSettings(),
+      RooFit::TestStatistics::LikelihoodMode{
+         static_cast<RooFit::TestStatistics::LikelihoodMode>(int(_cfg.parallelLikelihood))},
+      RooFit::TestStatistics::LikelihoodGradientMode::multiprocess, _cfg.verbose));
```

Comment:
```suggestion
   _fcn = std::make_unique<RooFit::TestStatistics::MinuitFcnGrad>(
      function, this, _theFitter->Config().ParamsSettings(),
      RooFit::TestStatistics::LikelihoodMode{
         static_cast<RooFit::TestStatistics::LikelihoodMode>(int(_cfg.parallelLikelihood))},
      RooFit::TestStatistics::LikelihoodGradientMode::multiprocess, _cfg.verbose);
```
You don't need to make this detour via the copy constructor. To `make_unique<T>`, you should directly pass the constructor arguments of `T`.